internal/pkg/endpoints/kubernetes_test.go

```diff
@@ -2184,6 +2184,117 @@ func TestPodTargetsPathLabel(t *testing.T) {
 	)
 }
 
+func TestPodTargetsPathAnnotationCorrectQuery(t *testing.T) {
+	t.Parallel()
+
+	assert.ElementsMatch(
+		t,
+		podTargets(&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-pod",
+				Namespace: "test-ns",
+				Annotations: map[string]string{
+					"prometheus.io/path": "/metrics?format=prometheus",
+				},
+			},
+			Spec: corev1.PodSpec{
+				NodeName: "node-a",
+				Containers: []corev1.Container{
+					{
+						Name: "app",
+						Ports: []corev1.ContainerPort{
+							{
+								Name:          "http-app",
+								ContainerPort: 80,
+							},
+						},
+					},
+				},
+			},
+			Status: corev1.PodStatus{
+				PodIP: "10.0.0.1",
+			},
+		}),
+		[]Target{
+			{
+				Name: "my-pod",
+				Object: Object{
+					Name: "my-pod",
+					Kind: "pod",
+					Labels: labels.Set{
+						"podName":        "my-pod",
+						"namespaceName":  "test-ns",
+						"deploymentName": "",
+						"nodeName":       "node-a",
+					},
+				},
+				URL: url.URL{
+					Scheme:   "http",
+					Host:     "10.0.0.1:80",
+					Path:     "/metrics",
+					RawQuery: "format=prometheus",
+				},
+			},
+		},
+	)
+}
+
+func TestPodTargetsPathAnnotationIncorrectQuery(t *testing.T) {
+	t.Parallel()
+
+	assert.ElementsMatch(
+		t,
+		podTargets(&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-pod",
+				Namespace: "test-ns",
+				Annotations: map[string]string{
+					"prometheus.io/path": "/metrics?format=prometheus?something",
+				},
+			},
+			Spec: corev1.PodSpec{
+				NodeName: "node-a",
+				Containers: []corev1.Container{
+					{
+						Name: "app",
+						Ports: []corev1.ContainerPort{
+							{
+								Name:          "http-app",
+								ContainerPort: 80,
+							},
+						},
+					},
+				},
+			},
+			Status: corev1.PodStatus{
+				PodIP: "10.0.0.1",
+			},
+		}),
+		[]Target{
+			{
+				Name: "my-pod",
+				Object: Object{
+					Name: "my-pod",
+					Kind: "pod",
+					Labels: labels.Set{
+						"podName":        "my-pod",
+						"namespaceName":  "test-ns",
+						"deploymentName": "",
+						"nodeName":       "node-a",
+					},
+				},
+				URL: url.URL{
+					Scheme: "http",
+					Host:   "10.0.0.1:80",
+					// This is fine, url.URL will escape this to /metrics%3Fformat%3Dprometheus%3Fsomething%0A
+					// upon making the request.
+					Path: "/metrics?format=prometheus?something",
```

Comment:
This sounds like a featurbug

Reply:
It was, such as putting query parameters in `prometheus.io/path` 😭 
I've reworked it to be more strict now.